src/pages/fr/core-concepts/endpoints.md

```diff
@@ -0,0 +1,280 @@
+
+Dans les sites générés statiquement, vos points de terminaison personnalisés sont appelés lors de la compilation afin de produire des fichiers statiques. Si vous optez pour le mode [SSR](/fr/guides/server-side-rendering/), points de terminaison personnalisés deviendront des points de terminaison "live server" qui seront appelés à la demande. Les points de terminaison Statiques et SSR sont définis de manière similaire, mais les points de terminaison SSR supportent des fonctionnalités additionnelles.
+
+## Points de terminaison statiques
+
+Pour créer un point de terminaison, ajoutez un fichier `.js` ou `.ts` dans le réppertoire `/pages`. L'extension `.js` or `.ts` sera supprimée lors du processus de compilation, pour que le nom de fichier inclus l'extension de données que vous souhaitez créer. Par exemple, `src/pages/data.json.ts` générera un point de terminaison `/data.json`.
+
+Les points de terminaison exportent une fonction `get` (optionellement `async`) qui reçoit un [objet context](/fr/reference/api-reference/#endpoint-context) avec des propriétés similaires à l'objet `Astro` global. Elle retourne un objet avec un `body`, et Astro l'appellera au moment de la compilation et utilisera les contenus du body pour générer le fichier.
+
+```ts
+// Exemple: src/pages/builtwith.json.ts
+// Génère: /builtwith.json
+export async function get({params, request}) {
+  return {
+    body: JSON.stringify({
+      name: 'Astro',
+      url: 'https://astro.build/',
+    }),
+  };
+}
+```
+
+L'objet retourné peut aussi avoir une propriété `encoding`. Cela peut être n'importe quel [`BufferEncoding`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bdd02508ddb5eebcf701fdb8ffd6e84eabf47885/types/node/buffer.d.ts#L169) valide accepté par la method Node.js' `fs.writeFile`. Par exemple, pour produire une image binaire png :
+
+```ts title="src/pages/astro-logo.png.ts" {6}
+export async function get({ params, request }) {
+  const response = await fetch("https://astro.build/assets/press/full-logo-light.png");
+  const buffer = Buffer.from(await response.arrayBuffer());
+  return {
+    body: buffer,
+    encoding: 'binary',
+  };
+}
+```
+
+Vous pouvez également saisir vos fonctions de points de terminaison en utilisant le type `APIRoute` :
+
+```ts
+import type { APIRoute } from 'astro';
+
+export const get: APIRoute = async function get ({params, request}) {
+...
+```
+
+### `params` et routage dynamique
+
+Les points de terminaison supportent le même fonctionnalités de [routes dynamiques](/fr/core-concepts/routing/routes-dynamiques) que les pages. Nommez votre fichier avec un nom de paramètre entre crochets et exportez une [fonction `getStaticPaths()`](/fr/reference/api-reference/#getstaticpaths). Ensuite, vous pouvez accéder au paramètre en utilisant la propriété `params` passé à la fonction du point de terminaison :
```

Comment:
Looks like this was missed!

```suggestion
Les points de terminaison supportent le même fonctionnalités de [routes dynamiques](/fr/core-concepts/routing/#routes-dynamiques) que les pages. Nommez votre fichier avec un nom de paramètre entre crochets et exportez une [fonction `getStaticPaths()`](/fr/reference/api-reference/#getstaticpaths). Ensuite, vous pouvez accéder au paramètre en utilisant la propriété `params` passé à la fonction du point de terminaison :
```

By the way, you should be able to commit these suggestions directly from the GitHub UI if you want to avoid needing to make the changes in your own editor.

Reply:
Ok, link + typo fixed.
Thank you, I know for the `Commit suggestion` feature.